wen-restart/src/wen_restart.rs

```diff
@@ -0,0 +1,150 @@
+//! The `wen-restart` module handles automatically repair in cluster restart
+
+use {
+    crate::solana::wen_restart_proto,
+    log::*,
+    prost::Message,
+    solana_gossip::cluster_info::ClusterInfo,
+    solana_ledger::blockstore::Blockstore,
+    solana_vote_program::vote_state::VoteTransaction,
+    std::{
+        fs::File,
+        io::{Error, Write},
+        path::PathBuf,
+        sync::Arc,
+    },
+};
+
+// The number of ancestor slots sent is hard coded at 81000, because that's
+// 400ms * 81000 = 9 hours, we assume most restart decisions to be made in 9
+// hours.
+const MAX_SLOTS_ON_VOTED_FORKS: u32 = 81000;
+
+pub fn wait_for_wen_restart(
+    wen_restart_path: &PathBuf,
+    last_vote: VoteTransaction,
+    blockstore: Arc<Blockstore>,
+    cluster_info: Arc<ClusterInfo>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    // repair and restart option does not work without last voted slot.
+    let last_vote_slot = last_vote.last_voted_slot().unwrap();
```

Comment:
nit: make this `expect` letting the user know if they've wiped their local tower they can't participate in wen_restart.

Reply:
Done.